include/openthread/tcp.h

```diff
@@ -610,14 +616,17 @@ typedef void (*otTcpAcceptDone)(otTcpListener *aListener, otTcpEndpoint *aEndpoi
  */
 struct otTcpListener
 {
-    struct otTcpListener *mNext;     ///< A pointer to the next TCP listener (internal use only)
-    otInstance *          mInstance; ///< A pointer to the OpenThread instance associated with this TCP listener
-    void *                mContext;  ///< A pointer to application-specific context
+    union
+    {
+        uint8_t mSize[16 + 3 * sizeof(void *)];
```

Comment:
Avoid magic numbers and comment on how this size was chosen?

Reply:
OK. In the latest update, I use `#define`s for these constants, preceded by a comment stating how the size was chosen (to match `struct tcpcb_listen` in this case). I did the same for the opaque `mTcb` field in `otTcpEndpoint`.